web/text_layer_builder.js

```diff
@@ -94,21 +92,24 @@ var TextLayerBuilder = (function TextLayerBuilderClosure() {
 
         var width = ctx.measureText(textDiv.textContent).width;
         if (width > 0) {
+          var textItem = textItems[i];
           textLayerFrag.appendChild(textDiv);
-          var transform;
-          if (textDiv.dataset.canvasWidth !== undefined) {
-            // Dataset values come of type string.
-            var textScale = textDiv.dataset.canvasWidth / width;
-            transform = 'scaleX(' + textScale + ')';
-          } else {
-            transform = '';
-          }
+          // Dataset values come of type string.
+          var canvasWidth = textItem.vertical ?
+              textItem.height * this.viewport.scale:
```

Comment:
Nit: remove two spaces from the indentation of this line and the next line. Each level is indented with two spaces in PDF.js.